pyg_lib/csrc/sampler/cpu/neighbor_kernel.cpp

```diff
@@ -134,25 +134,50 @@ class NeighborSampler {
 
     // Case 2: Sample with replacement:
     else if (replace) {
-      for (size_t i = 0; i < count; ++i) {
-        const auto edge_id = generator(row_start, row_end);
-        add(edge_id, global_src_node, local_src_node, dst_mapper,
-            out_global_dst_nodes);
+      if (row_end < (1 << 16)) {
```

Comment:
We could move this logic into `generator` instead of spilling it into the `sample` method.

Reply:
Thanks for the review :)
However there is a problem with the type of the array. Because when we use RandintEngine we don't know, which type of number it'll return (int32 or int64). So I would have to assume, that we have the worst case scenario (int64), which won't be optimal space-wise. 
Moreover, in this case, I would have to create 2 arrays - the one, that will store numbers generated by mkl (unfortunately, mkl can't take pointers / references to int32/int64 type arrays) and the array that will handle cases of int32/int64. Thus, I'll still need to check, which one of the arrays was used in neighbor_kernel.cpp